crates/matrix-sdk-base/src/store/send_queue.rs

```diff
@@ -324,6 +335,29 @@ pub struct SentMediaInfo {
     ///
     /// When uploading a thumbnail, this is set to `None`.
     pub thumbnail: Option<MediaSource>,
+
+    /// Accumulated list of infos for previously uploaded files and thumbnails
+    /// if used during a gallery transaction. Otherwise empty.
+    #[cfg(feature = "unstable-msc4274")]
+    #[serde(default)]
+    pub accumulated: Vec<AccumulatedSentMediaInfo>,
```

Comment:
There might be a different possible data scheme here, if I understand correctly:

- `SentMediaInfo` contains *only* the vec of `AccumulatedSentMediaInfo` (since a vec can hold the pair for the media, when there's only a single one and no gallery)
- `AccumulatedSentMediaInfo` (maybe rename `SingleSentMediaInfo`?) can keep on holding its current fields.

Again, I see this struct `SentMediaInfo` is also marked as `Serialized/Deserialize`, so maybe what I'm suggesting implies having to a data migration :woozy_face:.

Reply:
I think technically this would work, yes. I didn't go for it originally because I wanted to minimize breaking changes and because it felt slightly cleaner to have `accumulated` only contain the requests that are actually finished. This way, we only push to `accumulated`. If we do it the other way around, we'll need to push or modify the last item depending on what the request is for. Did you have specific advantages of using this variant in mind?

Reply:
The only advantage is technical, in that it avoids containing what's effectively another flattened `AccumulatedSentMediaInfo` next to the `accumulated` vector, and also reduces the number of concepts.

I'm a bit torn, because this is likely a non-trivial change, but on the other hand that *might* be a nice simplification. Maybe that could be attempted as a follow-up, and we could open an issue to not forget about it?